---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, echo = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/"
)
set.seed(0)
```

# Introduction

`DetectImports` is a R package aimed at distinguishing imported cases from locally acquired cases within a geographically limited genomic sample of an infectious disease. The input is a dated phylogeny of local genomes only, as can be build using `BEAST`, `treedater` or `BactDating`. The main output is an estimated probability of importation for each case in the dated phylogeny.

A scientific paper describing `DetectImports` in detail has been published:

Xavier Didelot, David Helekal, Michelle Kendall, Paolo Ribeca (2023). 
Distinguishing imported cases from locally acquired cases within a geographically limited genomic sample of an infectious disease.
Bioinformatics 39:btac761, https://doi.org/10.1093/bioinformatics/btac761

## Dependencies

`DetectImports` depends on [Stan](https://mc-stan.org/) through CmdStan. Although [other options might exist](https://mc-stan.org/docs/cmdstan-guide/cmdstan-installation.html), for instance using Conda, the simplest option is to install CmdStan via the R package [CmdStanR](https://mc-stan.org/cmdstanr/). You can do so with the commands
```{r eval = FALSE}
install.packages("cmdstanr", repos = c("https://mc-stan.org/r-packages/", getOption("repos")))
library(cmdstanr)
install_cmdstan()
```
The last command will download and compile all the missing dependencies.

## Installation

You can install `DetectImports` directly from github with the commands:

```{r gh-installation, eval = FALSE}
devtools::install_github("xavierdidelot/DetectImports")
```

The package can then be loaded using:
```{r}
library(DetectImports)
```

## Quick example

First we load and plot a dated tree stored in a Newick file. For this example we will use the file `example.nwk` which is distributed within the package `DetectImports`:

```{r}
library(ape)
path=system.file("extdata", "example.nwk", package = "DetectImports")
tree=read.tree(path)
plot(tree,show.tip.label=F)
axisPhylo(1)
```

We are now ready to detect the imports in this tree:

```{r message=F,warning=F,results='hide'}
library(DetectImports)
set.seed(0) #For reproducibility
res=detectImports(tree)
plot(res)
```

The bottom part is a scatter plot of the coalescent intervals for each sequence as a function of time, with the sequences corresponding to imports coloured according their probability. The top part shows the original tree with sequences coloured in red and sequences that have been propagating locally but descend from originally imported sequences coloured in blue.

## Advanced examples

For more advanced examples of use, see the [vignettes](https://github.com/xavierdidelot/DetectImports/tree/master/vignettes).

